src/common/exception/ErrorMessage.jl

```diff
@@ -0,0 +1,220 @@
+# This file is a part of GraknClient.  License is MIT: https://github.com/Humans-of-Julia/GraknClient.jl/blob/main/LICENSE 
+
+# Devide the ErrorClassification Client, Concept, Query and Internal is possible
+# with the following abstrat types. 
+abstract type AbstractGeneralError end
+abstract type AbstractClientError <: AbstractGeneralError end
+abstract type AbstractConceptError <: AbstractGeneralError end
+abstract type AbstractQueryError <: AbstractGeneralError end
+abstract type AbstractInternalError <: AbstractGeneralError end
+
+# The Symbol left side is assigned to the struct which will be build bellow
+# The right side of the Pair is assigned to the abstract type representing the section of error
+const error_structs = (
+    :Client_CLIENT_CLOSED=>:AbstractClientError,
+    :Client_SESSION_CLOSED=>:AbstractClientError, 
+    :Client_TRANSACTION_CLOSED=>:AbstractClientError, 
+    :Client_UNABLE_TO_CONNECT=>:AbstractClientError, 
+    :Client_NEGATIVE_VALUE_NOT_ALLOWED=>:AbstractClientError,  
+    :Client_MISSING_DB_NAME=>:AbstractClientError, 
+    :Client_DB_DOES_NOT_EXIST=>:AbstractClientError, 
+    :Client_MISSING_RESPONSE=>:AbstractClientError, 
+    :Client_UNKNOWN_REQUEST_ID=>:AbstractClientError, 
+    :Client_CLUSTER_NO_PRIMARY_REPLICA_YET=>:AbstractClientError, 
+    :Client_CLUSTER_UNABLE_TO_CONNECT=>:AbstractClientError, 
+    :Client_CLUSTER_REPLICA_NOT_PRIMARY=>:AbstractClientError, 
+    :Client_CLUSTER_ALL_NODES_FAILED=>:AbstractClientError, 
+    :Concept_INVALID_CONCEPT_CASTING=>:AbstractConceptError, 
+    :Concept_MISSING_TRANSACTION=>:AbstractConceptError,  
+    :Concept_MISSING_IID=>:AbstractConceptError, 
+    :Concept_MISSING_LABEL=>:AbstractConceptError, 
+    :Concept_BAD_ENCODING=>:AbstractConceptError, 
+    :Concept_BAD_VALUE_TYPE=>:AbstractConceptError, 
+    :Concept_BAD_ATTRIBUTE_VALUE=>:AbstractConceptError, 
+    :Query_VARIABLE_DOES_NOT_EXIST=>:AbstractQueryError, 
+    :Query_NO_EXPLANATION=>:AbstractQueryError,  
+    :Query_BAD_ANSWER_TYPE=>:AbstractQueryError,  
+    :Query_MISSING_ANSWER=>:AbstractQueryError,  
+    :Internal_UNEXPECTED_INTERRUPTION=>:AbstractInternalError, 
+    :Internal_ILLEGAL_STATE=>:AbstractInternalError, 
+    :Internal_ILLEGAL_ARGUMENT=>:AbstractInternalError, 
+    :Internal_ILLEGAL_CAST=>:AbstractInternalError)
+
+# building the error structs according the Pairs above with some meta programming
+for (T,A) in error_structs
+    @eval mutable struct $T <: $A
+        code_prefix::String  
+        code_number::Int 
+        message_prefix::String   
+        message_body::String
+    end
+end
+
+# All structs represent a unique number in a section of errors and the description. Both of them are 
+# assigned on the right side pair
+const error_messages = Dict([
+        Client_CLIENT_CLOSED=>( 1, "The client has been closed and no further operation is allowed."),
```

Comment:
Please unindent 4 spaces and similar style suggestions below.
```suggestion
        CLIENT_CLIENT_CLOSED => (1, "The client has been closed and no further operation is allowed."),
```